docs/src/operate/registration/register.md

```diff
@@ -108,39 +112,320 @@ GET /v1/entities
           "@type": "device"
       },
       {
-          "@topic-id": "device/main/service/tedge-agent",
+          "@topic-id": "device/main/service/service0",
           "@type": "service",
           "@parent": "device/main//"
       },
       {
           "@topic-id": "device/child0//",
           "@type": "child-device",
           "@parent": "device/main//"
-      },
-      {
-          "@topic-id": "device/child1//",
-          "@type": "child-device",
-          "@parent": "device/main//"
-      },
-      {
-          "@topic-id": "device/child00//",
-          "@type": "child-device",
-          "@parent": "device/child0//"
-      },
-      {
-          "@topic-id": "device/child01//",
-          "@type": "child-device",
-          "@parent": "device/child0//"
       }
+      ...
   ]
   ```
+* 404: Not Found
+  ```
+  Entity not found with topic id: device/unknown/topic/id
+  ```
```

Comment:
I'm not to sure about returning 404 when the query is empty.

Generally having no results in a query is not an error. Trying to access a resource that does not exist, would warrant a 404 response.

If you look at the REST API from Cumulocity, the query endpoints (e.g. `GET /inventory/managedObjects' don't return a 404 NOT FOUND HTTP status code when there are no matching results, it just returns an empty array. Technically speaking, I believe the "resource" in `GET /v1/entities` is '/v1/entities' itself...which does exist.

Reply:
Yeah, this was a confusing case for me as well and my original implementation was retuning empty array for non-existent entity. I changed it to 404 because empty array is returned for a leaf entity that doesn't have any children as well. So, when we get an empty array, we won't be able to differentiate if it's because there are no entities matching the query or if it's a non-existent entity. That's why I changed it to 404 for the latter case so that there is a clear distinction.

But, if the convention is empty array, then I'll change it back.

Reply:
Yes I would change it back to returning an empty array. This is much easier to deal with for people consuming the API as generally error handling generally wants to split "there are no results" from "the api request failed".

Reply:
And if you want to check the existence of a device, then you should be using `GET http://localhost:8000/tedge/entity-store/v1/entities/{entity}`

Reply:
Reverted that change in 1de9671a